packages/checkout-storefront/src/sections/CheckoutForm/useCheckoutSubmit.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  useCheckoutUpdateState,
+  useCheckoutUpdateStateActions,
+} from "@/checkout-storefront/state/updateStateStore";
+import {
+  useCheckoutValidationActions,
+  useCheckoutValidationState,
+} from "@/checkout-storefront/state/checkoutValidationStateStore";
+import { useCallback, useEffect, useState } from "react";
+import { useCheckoutFinalize } from "@/checkout-storefront/sections/CheckoutForm/useCheckoutFinalize";
+
+export const useCheckoutSubmit = () => {
+  const { validateAllForms } = useCheckoutValidationActions();
+  const { validating, validationState } = useCheckoutValidationState();
+  const { updateState, loadingCheckout } = useCheckoutUpdateState();
+  const { setShouldRegisterUser } = useCheckoutUpdateStateActions();
+  const { checkoutFinalize, finalizing } = useCheckoutFinalize();
+
+  const [submitInProgress, setSubmitInProgress] = useState(false);
```

Comment:
Wondering if it makes sense to move this value to the store as well, just to be consistent.

Then we could make a hook like `useIsCheckoutProcessing`

Reply:
I'd say no since it's only used in this one place and passed down to display the button loader so it seems to me more like a component state related than checkout itself